governance/eval-gate.py

```
@@ -0,0 +1,199 @@
+        except Red as e:
+            failures.append(f"{fam}: {key}: {e}")
+            continue
+        ratio = float(policy["regressions"][ratio_key])
+        if c > b * ratio:
+            failures.append(f"{fam}: 回归超上界（baseline={b} candidate={c} ratio_max={ratio}）")
+
+    # ---- contamination：数据集 digest 不得进优化输入面 ----
+    if "contamination" in required:
+        dg = (args.dataset_digest or "").strip()
+        if not dg:
+            die2("required 含 contamination 但 --dataset-digest 缺失（无从执法=fail-closed）")
```

Ruff (0.16.2)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)